pkg/storage/unified/apistore/managed.go

```diff
@@ -0,0 +1,80 @@
+package apistore
+
+import (
+	"net/http"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	authtypes "github.com/grafana/authlib/types"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+)
+
+func checkManagerPropertiesOnCreate(auth authtypes.AuthInfo, obj utils.GrafanaMetaAccessor) error {
+	kind := obj.GetAnnotation(utils.AnnoKeyManagerKind)
+	if kind == "" {
+		return nil
+	}
+
+	switch utils.ParseManagerKindString(kind) {
+	case utils.ManagerKindUnknown:
+		return nil // not managed
+
+	case utils.ManagerKindRepo:
+		if auth.GetUID() == "access-policy:provisioning" {
+			return nil // OK!
+		}
+		return &apierrors.StatusError{ErrStatus: metav1.Status{
+			Status:  metav1.StatusFailure,
+			Code:    http.StatusForbidden,
+			Reason:  metav1.StatusReasonForbidden,
+			Message: "Provisioned resources must be manaaged by the provisioning service account",
+		}}
+
+	case utils.ManagerKindPlugin, utils.ManagerKindClassicFP: // nolint:staticcheck
+		// ?? what identity do we use for legacy internal requests?
+		return nil // no error
+
+	case utils.ManagerKindTerraform, utils.ManagerKindKubectl:
+		manager, _ := obj.GetManagerProperties()
+		if manager.AllowsEdits {
+			return nil // no error anyone can do it
+		}
+		// TODO: check the kubectl+terraform resource
```

Comment:
@radiohead I am going to leave this as something we can implement soon.  The logic, constraints, testing and fallback is different enough that I don't feel comfortable trying to merge with this PR :)